src/error.rs

```diff
@@ -25,6 +26,9 @@ pub enum Error {
     /// io error
     #[display(fmt = "IO error: {}", _0)]
     Io(IoError),
+    /// signing error
+    #[display(fmt = "Signing error: {}", _0)]
+    Signing(String),
```

Comment:
Why not expose `EthsignError` instead?

Reply:
I tried but ran into some issues. `EthsignError` doesn't implement `Clone` or `PartialEq`. Let me take a look, maybe I can come up with something.

Reply:
I see. I don't really remember why we need both `Clone` and `PartialEq` here. If it's possible let's remove that if not, then could you file a separate issue for this, so that we can later change `String` to proper error?

Reply:
OK. That sounds like a good plan. There are other errors that are also converted to string that might benefit from having concrete error types as well (like `Error::Decoder`, `Error::Transport`).

Reply:
I don't know what the policy on defining error types in this crate is, but i can see that this crate doesn't use any special magic for its errors. Using something like `error-chain`+`error-chain-derive`, `derive-error`, `failure`, the new and shiny `anyhow` could make error definition and handling smoother.
(I think the ones i mention before `failure` are unmaintained)

Reply:
We've been actually getting rid of all the crates you've mentioned from other projects as they don't bring that much value. I'm a fan of a simple enum + `derive_more` to reduce boilerplate now.

Reply:
I think the biggest strength of those crates is the fact they provide backtraces for chains of errors. But of course i won't insist on this.

Reply:
Yeah, that's what you get if you implement `std::error::Error` and provide `source`. I agree that it's nice when automated, but as I said, I'm currently more inclined to just implement it manually given my bad experiences with both `error-chain` and `failure`.

Reply:
A little off-topic but I really like `thiserror`

Reply:
A little off-topic but I really like `thiserror`

Reply:
@nlordell Thanks for the hint, indeed looks very nice.